src/librustc_driver/driver.rs

```diff
@@ -1222,100 +1221,9 @@ where
             // tcx available.
             time(sess, "dep graph tcx init", || rustc_incremental::dep_graph_tcx_init(tcx));
 
-            time(sess, "looking for entry point", || {
-                middle::entry::find_entry_point(tcx)
-            });
-
-            time(sess, "looking for plugin registrar", || {
-                plugin::build::find_plugin_registrar(tcx)
-            });
-
-            time(sess, "looking for derive registrar", || {
-                proc_macro_decls::find(tcx)
-            });
-
-            time(sess, "loop checking", || loops::check_crate(tcx));
-
-            time(sess, "attribute checking", || {
-                hir::check_attr::check_crate(tcx)
-            });
-
-            time(sess, "stability checking", || {
-                stability::check_unstable_api_usage(tcx)
-            });
-
-            // passes are timed inside typeck
-            match typeck::check_crate(tcx) {
-                Ok(x) => x,
-                Err(x) => {
-                    f(tcx, rx, Err(x));
-                    return Err(x);
-                }
-            }
-
-            time(sess, "rvalue promotion", || {
-                rvalue_promotion::check_crate(tcx)
-            });
-
-            time(sess, "privacy checking", || {
-                rustc_privacy::check_crate(tcx)
-            });
-
-            time(sess, "intrinsic checking", || {
-                middle::intrinsicck::check_crate(tcx)
-            });
+            tcx.analysis(LOCAL_CRATE).ok();
 
```

Comment:
This should return the error, shouldnt't it.

Reply:
We pass the compile status to the closure and expect it to handle it. I'll change `phase_3_run_analysis_passes` to return just `R`.